### ISSUE TITLE
fixtures: fix loading of record 1395

### ISSUE THREAD

**Issue by tiborsimko** [cernopendata/opendata.cern.ch#1826]
Record 1395 from `cms-simulated-datasets-Run2011A.json` contains ~15000 files, which seems to cause troubles when loading it with files from the fixtures.  

> elasticsearch.exceptions.ConnectionError: ConnectionError(HTTPConnectionPool(host='search', port=9200): Max retries exceeded with url: /records-record-v1.0.0/record-v1.0.0/c4ff9928-6310-4489-8a67-dc149250a240?version=0&version_type=external_gte (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x5f4db0d0>: Failed to establish a new connection: [Errno 111] Connection refused',))) caused by: ConnectionError(HTTPConnectionPool(host='search', port=9200): Max retries exceeded with url: /records-record-v1.0.0/record-v1.0.0/c4ff9928-6310-4489-8a67-dc149250a240?version=0&version_type=external_gte (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x5f4db0d0>: Failed to establish a new connection: [Errno 111] Connection refused',)))

Investigate more permissive timeouts.

Investigate switching to bulk batch indexing.

**Comment by tiborsimko** [comment]
See how it looks like in COD2:

- http://opendata.cern.ch/record/1395

**Comment by tiborsimko** [comment]
Sort of good news 😄 Following up #1831, I could load the record locally. However, it takes about ~10 minutes to load this big record on my laptop, and the display is slow...

**Comment by pamfilos** [comment]
closing as issue fixed in https://github.com/cernopendata/opendata.cern.ch/pull/1874/commits/bac93a6966fe19af4cc51de3e4586e6bb130b897

**Comment by tiborsimko** [comment]
@pamfilos I'm not sure your commit closes the issue, since the primary problem is with the *record loading* at the fixture time, not with the "record displaying" once it is successfully loaded... It still takes ~10 minutes to load this one record on my laptop:

```
real	7m30.295s
user	2m34.846s
sys	0m6.416s
```
Taking into accont that we shall have bigger records in 2012 MC datasets, this may be easily back...

**Comment by tiborsimko** [comment]
@pamfilos Reopening this issue. You can create `cod3-record-1395.json` for a good example, taking recid=1395 from `cms-simulated-datasets-Run2011A.json`, and load it via:

```
$ cernopendata fixtures records \
    -f ./cernopendata/modules/fixtures/data/records/cod3-record-1395.json \
    --mode insert
```

It should take ~10 minutes on your laptop, so this is the reference time.  

The goal is to speed this up, possibly via option No.1 in PR #1759, possibly via CDS-like technique:

-  https://github.com/inveniosoftware/invenio-files-rest/tree/cds-object-version-tags  
- https://github.com/CERNDocumentServer/cds-videos/blob/cdslabs_qa/requirements.topical.branches.txt

**Comment by tiborsimko** [comment]
@pamfilos Here's another option how to handle dataset records that contain ~15,000 files:

- nomenclature: there are about ~15 "index files" (=text files with URIs to ROOT files) and about ~15,000 "asset files" (=ROOT files themselves)

- we can load only the "index files" together with the record, and send these to DB and ES

- the file pagination box could use index file content, which is basicaly a detailed list of all the asset files that we should show to users, and paginate over the files listed there

- however, the index files contain only the file location and name, not the file size; if we want to show the file size information, which I think is mandatory, we need to create "richer" index files

- these richer index files would be in JSON format, copying the current fixture structure, and stored on EOS alongside the current index files, for example:

   - `root://.../file-indexes/CMS_Run2012B_BJetPlusX_AOD_22Jan2013-v1_20000_file_index.txt` -- the old plain text structure with URIs to asset files, like now` used by CMS analyses etc
   - `root://.../file-indexes/CMS_Run2012B_BJetPlusX_AOD_22Jan2013-v1_20000_file_index.json` -- a new richer JSON structure about asset files, with URIs and sizes and checksums, as we have them in the fixtures right now

- the file pagination box could fetch the new richer index files by calling `/record/NNN/files/XXX.json` which would fetch them from EOS (and which would also populate local NGINX cache for faster accesses) and offer the file iteration box based on the information they contain 

- we could implement this using the same `files` property, with fixtures containing:
  - missing `files.type` would indicate asset files, to be added to record and sent to DB and to ES
  - `files.type=index` would indicate current index index files, to be added to record and sent to DB and to ES
  - `files.type=indexjson` would indicate the new richer index files in JSON, to be added to record and sent to DB and to ES
  - fixtures wouldn't even have to contain the individual files, as these would be stored in the file index files on EOS; alternatively, if we want to keep the information in the fixture files for safety reasons, which is perhaps a good thing, we can list them under a new `files_hidden` property, or other such better name, and they won't be added to the record and sent neither to DB nor to ES

- so, to sum up how the fixtures would be treated by the fixture loader: 
  - everything under `files` would be stored with the record and send to DB and ES
  - everything under `files_hidden` (or other such better name) would be ignored

WDYT?

**Comment by tiborsimko** [comment]
Fixed by the removal of individual asset files, making the upload fast, and by amending the detailed record page view to offer a "List files" functionality working with rich JSON files.
